errai-bus/src/main/java/org/jboss/errai/bus/client/framework/ClientMessageBusImpl.java

```diff
@@ -948,25 +948,44 @@ private void sendAllDeferred() {
     final List<Message> highPriority = new ArrayList<>();
     for (final Message message : new ArrayList<>(deferredMessages)) {
       if (message.hasPart(MessageParts.PriorityProcessing)) {
-        if (remotes.containsKey(message.getSubject()))
+        if (remotes.containsKey(message.getSubject())) {
           highPriority.add(message);
-        deferredMessages.remove(message);
+          deferredMessages.remove(message);
+        }
       }
     }
 
     final List<Message> lowPriority = new ArrayList<>();
-    do {
+    for(Message m : deferredMessages) {
```

Comment:
I don't think we need this outer loop anymore, and in fact, it is causing a ConcurrentModificationException (because of the deferredMessage.remove call) that is the source of some test failures.

Reply:
True you are right, lack of sleep makes for A+ coding haha, will fix that up asap